internal/plugin/plugin.go

```diff
@@ -81,7 +81,9 @@ func Walk(dir string, f func(plugin *Plugin)) error {
 		return err
 	}
 	for _, p := range sorted {
-		f(p)
+		if err := f(p); err != nil {
+			return err
```

Comment:
I had a fair amount of logic while walking plugins and wanted to fail fast, so updated this method signature to allow it.